index.bs

```diff
@@ -3406,6 +3421,69 @@ and |session|:
 
 </div>
 
+<div algorithm="collect nodes using accessibility attributes">
+To <dfn>collect nodes using accessibility attributes</dfn> with given |context nodes|, |selector|,
+|maximum returned node count|, and |returned nodes|:
+
+1. If |returned nodes| is null:
+
+  1. Set |returned nodes| to an empty [=/list=].
+
+1. For each |context node| in |context nodes|:
+
+  1. Let |match| be true.
+
+  1. If |selector| [=map/contains=] "<code>role</code>":
+
+    1. Let |role| be the [=computed role=] of |context node|.
+
+    1. If |selector|["<code>role</code>"] [=is|is not=] |role|:
+
+      1. Set |match| to false.
+
+  1. If |selector| [=map/contains=] "<code>name</code>":
+
+    1. Let |name| be the [=accessible name=] of |context node|.
+
+    1. If |selector|["<code>name</code>"] [=is|is not=] |name|:
+
+      1. Set |match| to false.
+
+  1. If |match| is true:
+
+    1. If |maximum returned node count| is not null and [=list/size=] of |returned
+      nodes| is greater or equal to |maximum returned node count|:
+
+      1. Return [=success=] with data |returned nodes|.
+
+    1. [=list/Append=] |context node| to |returned nodes|.
+
+  1. Let |child nodes| be an empty [=/list=] and, for each node |child| in the
+     <a spec=dom>children</a> of |context node|:
+
+    1. If |child| implements {{Element}}, [=list/append=] |child| to |child nodes|.
+
+  1. [=Try=] to [=collect nodes using accessibility attributes=] with |child nodes|,
+     |selector|, |maximum returned node count|, and |returned nodes|.
+
+  1. Return [=success=] with data |returned nodes|.
```

Comment:
This is indented too much, right? i.e. the return should be after the loop, otherwise we only ever process the first context node and the first element children below it.